### PR TITLE
[WIP] Running travis

### PR DESCRIPTION
**Pull request by nickvergessen**
Automated test case testing :)

todo:
- [ ] Revert `lib/test_basicSync.py`
- [ ] Revert `python/smashbox/utilities/__init__.py`
- [ ] Test stable8 and master

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+branches:
+  only:
+    - master
+    - /^stable\d*$/
+
+env:
+  global:
+    - CORE_BRANCH=stable8
+    - APP_NAME=smashbox
+    - DB=sqlite
```
Comment (nickvergessen) [inline .travis.yml:16]: Should we test all DBs?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
+language: php
+
+matrix:
+  include:
+    - php: 5.5
```
Comment (nickvergessen) [inline .travis.yml]: Which php versions should we check?

I don't want a too huge testing matrix

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+script:
+
+  # Run smashbox tests
+  - bin/smash lib/$TEST_NAME.py
+
+after_script:
+  # Output the logs
+  - cat ../../data/owncloud.log
+  # FIXME longer then 10k lines, travis asks you to not output that
+  # - cat ~/smashdir/log-test_basicSync.log
```
Comment (nickvergessen) [inline .travis.yml:97]: Would be nice if we can filter output the file, so we get errors and stuff

File: lib/test_basicSync.py [hunk 4]
```diff
@@ -57,6 +57,15 @@
 ]
 
 def expect_content(fn,md5):
+    logger.warning('DEBUGGING expect_content: ' + fn)
```
Comment (nickvergessen) [inline lib/test_basicSync.py]: Revert

File: lib/test_basicSync.py [hunk 5]
```diff
@@ -101,6 +110,10 @@ def creator(step):
     step(1,'create initial content and sync')
 
     d = make_workdir()
+    cmd = 'ls -l %s'
```
Comment (nickvergessen) [inline lib/test_basicSync.py]: Revert

File: python/smashbox/utilities/__init__.py [hunk 6]
```diff
@@ -279,7 +279,10 @@ def runcmd(cmd,ignore_exitcode=False,echo=True,allow_stderr=True,shell=True,log_
 
     if echo:
         if stdout.strip():
-            logger.info("stdout: %s",stdout)
+            if not (('ls -l ' in cmd) or ('md5' in cmd) or ('&& echo "Found" || echo "Not found"' in cmd)):
```
Comment (nickvergessen) [inline python/smashbox/utilities/__init__.py]: Revert

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  fast_finish: true
+
+branches:
+  only:
+    - master
+    - /^stable\d*$/
+
+env:
+  global:
+    - CORE_BRANCH=stable8
```
Comment (nickvergessen) [inline .travis.yml]: Decide which branch(es) to test by default

File: .travis.yml (hunk 2 above)
Comment (DeepDiver1975) [inline .travis.yml]: we shall at least test the min req - 5.4

File: .travis.yml (hunk 7 above)
Comment (DeepDiver1975) [inline .travis.yml]: so the intention  here is to test smashbox or owncloud by running smashbox?

From my understanding the first - master or latest stable

**Comment by DeepDiver1975** [comment]
new pr - https://github.com/owncloud/smashbox/pull/86

**Pull request closed without merging**